src/v/coding-style.md

```diff
@@ -6,12 +6,50 @@ The document covers style and convention not enforced by clang-format.
 
 Header files have the `.h` extension, source files use the `.cc` extension.
 
+Public header files should go in an `includes` directory within the module they belong to. 
```

Comment:
This may need some clarification of what "public" means here. Is "public" is entirely concerned with whether other redpanda modules #include them directly?

Reply:
Yes. Will clarify with:

```
Public header files (files that are allowed to be #included by other modules).
```